packer/plugin-getter/plugins.go

```diff
@@ -170,30 +189,35 @@ func (pr Requirement) ListInstallations(opts ListInstallationsOptions) (InstallL
 			continue
 		}
 
-		matches := pluginVersionRegex.FindStringSubmatch(pluginVersionStr)
-		if matches == nil {
-			log.Printf("invalid version found: %q, ignoring", pluginVersionStr)
+		rawVersion, err := version.NewVersion(pluginVersionStr)
+		if err != nil {
+			log.Printf("malformed version string in filename %q: %s, ignoring", pluginVersionStr, err)
 			continue
 		}
 
-		absVersion := matches[1]
-		if len(matches) == 3 {
-			absVersion = fmt.Sprintf("%s%s", absVersion, matches[2])
+		descVersion, err := version.NewVersion(describeInfo.Version)
+		if err != nil {
+			log.Printf("malformed reported version string %q: %s, ignoring", describeInfo.Version, err)
+			continue
 		}
 
-		if absVersion != describeInfo.Version {
-			log.Printf("plugin %q reported version %s while its name implies version %s, ignoring", path, describeInfo.Version, absVersion)
+		if rawVersion.Compare(descVersion) != 0 {
+			log.Printf("plugin %q reported version %q while its name implies version %q, ignoring", path, describeInfo.Version, pluginVersionStr)
 			continue
 		}
 
-		rawVersion, _ := version.NewVersion(matches[1])
+		preRel := descVersion.Prerelease()
```

Comment:
The regex ensures that the only prerelease flag allowed is `dev` by using the version Package to validate if a version is valid we remove this validation and allow all versions. But this breaks our validation here as `1.1.1+metadata-dev` is not a dev release but an official release with metadata. Calling `version.Prerelease()` on this version returns an empty version string, which means user could install plugins with different types of preelease stages, which is not what we want right? 

https://go.dev/play/p/Mb0ZMFj6YdT

Reply:
Update: after discussing this out-of-band, we opted to scrub the metadata information from the version.
We won't attempt to correct/reject the version if something like `1.1.1+metadata-dev` is pushed since we don't know for a fact this wasn't intended, even if the expected format would be `1.1.1-dev+metadata`.
This change is mostly to avoid conflicts with multiple instances of the same version installed: metadata is not considered part of the version for comparison purposes, so by removing it from the filename, we make sure we'll only have one instance of a plugin at a given version locally.